modules/core/shared/src/main/scala/lucuma/core/data/PerSite.scala

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2016-2025 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.core.data
+
+import cats.Functor
+import cats.kernel.Eq
+import cats.kernel.Monoid
+import cats.syntax.monoid.*
+import io.circe.Decoder
+import io.circe.Encoder
+import io.circe.Json
+import io.circe.syntax.*
+import lucuma.core.enums.Site
+
+/** A total map from Site to A. */
+final case class PerSite[+A](gn: A, gs: A) extends (Site => A):
+
+  /** PerSite is a total function from Site to A. Alias for `forSite`. */
+  def apply(site: Site): A =
+    forSite(site)
+
+  /** Lower to a pair. */
+  def toPair: (A, A) =
+    (gn, gs)
+
+  /** Select the value for the given site. Alias for `apply`. */
+  def forSite(site: Site): A =
+    site match
+      case Site.GN => gn
+      case Site.GS => gs    
+
+  /** PerSite is a functor. */
+  def map[B](f: A => B): PerSite[B] =
+    PerSite(f(gn), f(gs))
+
+  /** Examine each (Site, A) pair and fold into monoid B. */
+  def fold[B: Monoid](f: (Site, A) => B): B =
```

Comment:
Maybe a `fold(gn: A => B, gs: A => B)` would be useful.

Reply:
I thought about that signature but it's kind of like Boolean.fold ... there's nothing to prevent you from passing the args in the wrong order. I'll add it and change this one to `foldWithSite` or something.